replica/database.cc

```diff
@@ -896,12 +897,16 @@ future<> database::update_keyspace(sharded<service::storage_proxy>& proxy, const
 }
 
 future<> database::update_keyspace_on_all_shards(sharded<database>& sharded_db, sharded<service::storage_proxy>& proxy, const sstring& name) {
-    return modify_keyspace_on_all_shards(sharded_db, [&] (replica::database& db) {
-        return db.update_keyspace(proxy, name);
-    }, [&] (replica::database& db) {
-        const auto& ks = db.find_keyspace(name);
-        return db.get_notifier().update_keyspace(ks.metadata());
-    });
+    try {
+        co_await modify_keyspace_on_all_shards(sharded_db, [&] (replica::database& db) {
+            return db.update_keyspace(proxy, name);
+        }, [&] (replica::database& db) {
+            const auto& ks = db.find_keyspace(name);
+            return db.get_notifier().update_keyspace(ks.metadata());
+        });
+    } catch (...) {
+        on_fatal_internal_error(dblog, fmt::format("Failed updating keyspace {}: {}", name, std::current_exception()));
+    }
```

Comment:
> 
> We don't do that for create/drop keyspace since
> this affects only that keyspace that is either not used yet
> (on create) or not used anymore (in the drop case).
> FWIW, the error is now propagated upstream.

But this still leaves the node in a bad state, which for example will make the recreation with the same name fail, if we ignore the error. But do we really ignore it? The exception will propagate to the group0 apply fiber and halt the group0 state machine. In that case, catching exceptions here is also unnecessary.

Reply:
okay.  I wasn't aware of the group0 handling.
That said, if we decided to backport this series, older branches may need this catch here.

Reply:
dropped this try/catch in v4